homeassistant/components/stream/__init__.py

```diff
@@ -507,22 +521,28 @@ def _should_retry() -> bool:
     {
         vol.Optional(CONF_RTSP_TRANSPORT): vol.In(RTSP_TRANSPORTS),
         vol.Optional(CONF_USE_WALLCLOCK_AS_TIMESTAMPS): bool,
+        vol.Optional(CONF_EXTRA_PART_WAIT_TIME): cv.positive_float,
     }
 )
 
 
-def convert_stream_options(stream_options: dict[str, str | bool]) -> dict[str, str]:
+def convert_stream_options(
+    hass: HomeAssistant, stream_options: Mapping[str, str | bool | float]
+) -> tuple[dict[str, str], StreamSettings]:
     """Convert options from stream options into PyAV options."""
+    stream_settings = copy.copy(hass.data[DOMAIN][ATTR_SETTINGS])
     pyav_options: dict[str, str] = {}
     try:
         STREAM_OPTIONS_SCHEMA(stream_options)
     except vol.Invalid as exc:
         raise HomeAssistantError("Invalid stream options") from exc
 
+    if extra_wait_time := stream_options.get(CONF_EXTRA_PART_WAIT_TIME):
+        stream_settings.hls_part_timeout += extra_wait_time
     if rtsp_transport := stream_options.get(CONF_RTSP_TRANSPORT):
         assert isinstance(rtsp_transport, str)
         pyav_options["rtsp_transport"] = rtsp_transport
     if stream_options.get(CONF_USE_WALLCLOCK_AS_TIMESTAMPS):
         pyav_options["use_wallclock_as_timestamps"] = "1"
 
-    return pyav_options
+    return pyav_options, stream_settings
```

Comment:
It feels like `create_stream` should accept stream_options, then it can split these into pyav options and stream settings internally.

Reply:
Not quite sure I follow...isn't that what it's doing now? `convert_stream_options` is just used within `create_stream`. Maybe we can make it an inner function to make this more clear